extensions/amp-image-slider/0.1/test/integration/test-amp-image-slider.js

```diff
@@ -446,5 +470,95 @@ config.run('amp-image-slider', function() {
+            button = doc.querySelector('#button');
+            pos10Percent = rect.left + 0.1 * rect.width;
+            button.click();
+            return pollByBarLeftPos(pos10Percent);
+          });
+    });
+
+    it('should show hint again on scroll back and into viewport', () => {
+      let slider, container_, hint;
+      let mouseDownEvent;
+      return prep()
+          .then(() => {
+            slider = doc.querySelector('#slider');
+            container_ = slider
+                .querySelector('.i-amphtml-image-slider-container');
+            hint = slider.querySelectorAll('.i-amphtml-image-slider-hint')[0];
+            mouseDownEvent = createFakeEvent(
+                'mousedown',
+                0,
+                0
+            );
+            container_.dispatchEvent(mouseDownEvent);
+            return poll('should receive hidden class', () => {
+              return hint.classList
+                  .contains('i-amphtml-image-slider-hint-hidden');
+            }, () => {});
+          })
+          .then(() => {
+            env.win.scrollTo({
+              top: 3000,
+            });
+            // Have to use timeout here,
+            // no indication of proper viewportCallback trigger
+            return timeout(env.win, 500);
+          })
+          .then(() => {
+            env.win.scrollTo({
+              top: 0,
+            });
+            return poll('should remove hidden class', () => {
+              return !hint.classList
+                  .contains('i-amphtml-image-slider-hint-hidden');
+            }, () => {});
+          });
+    });
+
+    it('should show hint again with disable-hint-reappear', () => {
+      let slider, container_, hint;
+      let mouseDownEvent;
+      return prep()
+          .then(() => {
+            slider = doc.querySelector('#s2');
+            container_ = slider
+                .querySelector('.i-amphtml-image-slider-container');
+            hint = slider.querySelectorAll('.i-amphtml-image-slider-hint')[0];
+            mouseDownEvent = createFakeEvent(
+                'mousedown',
+                0,
+                0
+            );
+            container_.dispatchEvent(mouseDownEvent);
+            return poll('should receive hidden class', () => {
+              return hint.classList
+                  .contains('i-amphtml-image-slider-hint-hidden');
+            }, () => {});
+          })
+          .then(() => {
+            env.win.scrollTo({
+              top: 3000,
+            });
+            // Have to use timeout here,
+            // no indication of proper viewportCallback trigger
+            return timeout(env.win, 500);
+          })
+          .then(() => {
+            env.win.scrollTo({
+              top: 0,
+            });
+            return poll('should remove hidden class', () => {
+              return hint.classList
+                  .contains('i-amphtml-image-slider-hint-hidden');
+            }, () => {});
```

Comment:
Unclear why you are passing a noop function as the second argument. It looks like by default, the error message will be 'Timeout waiting for should remove hidden class', but if you pass this, the error message will be empty.

Reply:
Forgot to remove this, was from original test impl. Will remove